### ISSUE TITLE
Create issue response tracking script

### ISSUE THREAD

**Issue by azaroth42** [archesproject/arches#4410]
One of the outcomes of a meeting between Getty Digital, Getty Conservation, Farallon and Open Tech Strategies was an action to create a script that lists issues that need attention.

The basic description was issues that have been open for a week without a comment on them. This isn't really sufficient, as there are lots of situations that would fail this test but still need a response (e.g. only comments from the creator, such as "heh, anyone out there?" or updates to the report) or when there are commits that reference the issue with labels assigned by someone other than the creator, and thus it's clearly being worked on.

This is a meta issue to decide the features that should be tracked, and whether the list should be time ordered, priority ordered and so forth.

An initial proposal:

Assign each issue a weight based on the following criteria, and sort the list by descending weight.

* No comments from someone with commit status: +25
* Created by someone without commit status: +10
* No reactions: +5
* No labels: +5
* Body includes image or URL: +5
* Title or label includes "bug": +5
* References from commits or other issues: -5
* Has labels from someone other than creator: -5
* Has comment from someone with commit status: -10
* Referenced from a PR: -10
* Has a milestone: -10
* Has an assignee: -20

Tagging: @dwuthrich @workergnome

Thoughts?

**Comment by dwuthrich** [comment]
@azaroth42 Agreed that the "open for a week" condition probably isn't sufficient.  I like a ranking system that attempts to weight the "severity" of the ticket, and you've provided a nice set of weighting criteria.

I'd personally default to the fewest number of features to track, and the 12 you suggest seem more than sufficient.

**Comment by mradamcox** [comment]
We already have a set of labels that we use for priority level--Low, Medium, High--shouldn't these be included in the metric?

**Comment by azaroth42** [comment]
@mradamcox Apologies for the lack of background discussion for the issue. The intent isn't to prioritize /work/ but instead to ensure that issues that are created get responses in a timely fashion, relative to their "weight". Even if that response is "sorry, won't fix" or that it's a low priority to be assigned to a backlog, that's sufficient to put it to the bottom of the issue triage list.

This was in reaction to noting that there are currently 296 out of 417 issues that do not have any comments on them, and of those, 77 do not even have any labels. That's a lot of issues to even catch up with, so we want to approach it in a systematic and highest impact first kind of way.  By consistently keeping up with high priority issues, we encourage engagement from the broader community rather than just the core players.

**Comment by mradamcox** [comment]
I see, so it's not necessarily a script to rank which tickets should be addressed with new code, but which should be addressed from the repo side of things (before any new code is written). For what it's worth, the ranking system looks like a good starting point for me.

**Comment by azaroth42** [comment]
Another factor -- is a pull request:  +20

PRs demonstrate real attention to the issue and code, and should be addressed and accepted or closed sooner rather than later. Otherwise they'll go stale as the codebase shifts around them.

Relatedly, I think the dependabot PRs should either be closed or accepted with some priority at this stage. They're making it very hard to see "real" PRs. If the bot is valuable, then it should be used. If not, then just get rid of it and the PRs it created.

**Comment by mradamcox** [comment]
I completely agree on dependabot. In planning for this year I was hoping to have some time to address more of those PRs personally, but I don't see that happening soon unfortunately. Perhaps it would be good to put out a call specifically for community help with those PRs? Or to see if any current contributors could commit some time to them. Not every dependabot PR is the same, either... Some that are micro version updates are very likely to pass all tests without any trouble. Others jump many versions, and will require more scrutiny.

**Comment by azaroth42** [comment]
Notes from review session between @dwuthrich and @azaroth42:

* If a PR can't be merged because there's nothing to do (eg #4429) then ask the submitter to close or fix the PR. Could be that the PR got messed up, could be that another merge fixed the issue, but closing without interaction could be a negative experience. Assign back to the submitter.
* If a PR has reviewers assigned but no reviews `@`nudge them to do the review
* Triagers should comment with references to related issues if possible.
* If an issue is determined to be a valid bug, then assign to the Validated Issues milestone. If it cannot be validated due to lack of information, assign to Information Requested milestone.

**Comment by azaroth42** [comment]
We got through 16 issues in an hour, but that involved a lot of discussion about the process, creating new labels and milestones and so on.  Assuming somewhere between 20 and 30 issues in an hour are possible to triage while pairing.
